src/app/shared/models/deserializable.model.ts

```diff
@@ -0,0 +1,3 @@
+export interface Deserializable {
```

Comment:
FYI: In TypeScript exist Class Abstract https://www.typescriptlang.org/docs/handbook/classes.html#abstract-classes.

This comment is not a required change, don't worry, keep with this code :)

Reply:
O maybe yes, because all the classes have the same implementation for deserialize

Reply:
Please contact me to review that

Reply:
You're damn right, done.